.github/workflows/cd-app.yml

```diff
@@ -14,14 +14,18 @@ on:
   workflow_dispatch:
     inputs:
       environment:
-        description: "target environment"
+        description: Environment to be deployed to
         required: true
         default: "dev"
         type: choice
         options:
           - dev
           - staging
           - prod
+      version:
```

Comment:
Was between `ref` and `version`. Went with `version` as I think this workflow is a little more non-technical user facing, and for the odd chance "versions" in some project may not correspond to something directly in git (would be pretty weird) and so they might want to update this to do a little translation before passing on to the actual deploy job. But no strong opinion here.

I do wonder if the other inputs added to the "lower level" workflows should maybe use `ref`? Like we do in build-and-publish?

And related side note, I noticed some workflows use `commit_hash` as input name for largely the same purpose. Should we standardize on one naming convention for "git reference thing"? Or is the naming there intentional that those other workflows specifically require a full commit SHA?

Reply:
Sounds good with the choice to use `version`. Agree that having a name that reflects semantics of the release process rather than the underlying implementation (git refs) makes sense here.

deploy.yml and database-migrations.yml are callable workflows, but in my mind they are still relatively high level workflows, whereas build-and-publish.yml is lower level. Somewhat subjective but I'm fine keeping it the way you have it already here.

re: flows that use `commit_hash`, yeah those flows require the full Sha. A ref won't work. The commit_hash is used directly to set the terraform variable image_tag in the terraform apply command. That flow explicitly doesn't reuse the deploy workflow because we want to bypass running database migrations, which would be hairy trying to support on PR environments since all PR environments share the database with the dev environment.